spec/lang/well-formed.md

```diff
@@ -444,7 +510,9 @@ impl Statement {
             Assign { destination, source } => {
                 let left = destination.check_wf::<T>(func.locals, prog)?;
                 let right = source.check_wf::<T>(func.locals, prog)?;
+                // FIXME: to fix 2 tests, I would need an exception here for `Ptr(PtrType::Raw)` where the layout doesn't matter.
                 ensure_wf(left == right, "Statement::Assign: destination and source type differ")?;
```

Comment:
This is the thing that now evaluates differently, since the raw pointer has to carry around its layout aswell (or at least the metadata kind, which is not a thing right now)